app/javascript/dashboard/routes/dashboard/settings/inbox/channels/Sms.vue

```diff
@@ -0,0 +1,145 @@
+<template>
```

Comment:
This component and Whatsapp component have a lot of items in common. It feel unnecessary to have logic in 2 places even though the API call is same with a change in the type of integration. Can we create a common component to handle these?

Reply:
@pranavrajs I duplicated code for SMS and Whatsapp component because when Chatwoot supports new API providers apart from Twilio like Interakt, Gupshup, Wednesday etc, there will be a lot of uncommon form fields between the two components and will vary from provider to provider. For this reason alone, I have duplicated most of the logic in the component because very soon there will be a lot of changes between SMS and Whatsapp depending on the provider.

Reply:
@srameshr That's fair argument to keep the duplicated code. But the scenario is too far to happen IMHO.
How about we create a Form component to render the form which is same for both and can re used and once field options for a channel changes we could create another component for the new set of fields. In this way it'll be simple to maintain the codebase.

Reply:
I totally understand that but seeing how many new Twilio competitors focused on WhatApp who are more affordable are popping up, I am sure that requirement as a feature your enterprise clients need is very near. Even the community on discord keeps asking for new providers to be supported.

Having said this, do you think we need to do something like this:
1. Have two separate channels
2. Render the same common component (something like SMSAndWhatsapp.js)for both the channels but with a prop passed to differentiate the type and also the translation messaging?
3. In the future, provide another form option called ```Provider``` within a select that renders the fields required by the ```Provider```?

Reply:
@srameshr Let us keep Sms.vue and Whatsapp.vue as separate files, but we can create a new component for Twilio actions, use that component in these places.

This way if you want to switch the provider you just need to swap the child component.

Reply:
@pranavrajs I've done that. Twilio is a common component with common fields and actions for both SMS and WhatsApp.